common/djangoapps/enrollment/views.py

```diff
@@ -783,3 +789,88 @@ def post(self, request):
                     actual_activation=current_enrollment['is_active'] if current_enrollment else None,
                     user_id=user.id
                 )
+
+
+@can_disable_rate_limit
+class EdraakCourseListView(APIView, ApiKeyPermissionMixIn):
+    """
+    """
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser,
+                              EnrollmentCrossDomainSessionAuth,)
+    permission_classes = ApiKeyHeaderPermissionIsAuthenticated,
```

Comment:
If we're showing just a list of courses to all learners, I'm not sure why authentication is needed at all.

```suggestion
    authentication_classes = []
    permission_classes = []
```

Reply:
It should show courses for (currently logged-in user) only. It doesn't have a `user` filter. Can you please check again?

Reply:
Nope let's keep the permissions since it's a per-user thing. I didn't understand the API so looks like we need to drop the caching.